src/sentry/api/endpoints/user_emails.py

```diff
@@ -0,0 +1,240 @@
+from __future__ import absolute_import
+
+import logging
+
+from django.db import IntegrityError, transaction
+from django.db.models import Q
+from rest_framework import serializers, status
+from rest_framework.response import Response
+
+from sentry import newsletter
+from sentry.api.bases.user import UserEndpoint
+from sentry.api.decorators import sudo_required
+from sentry.api.serializers import serialize
+from sentry.models import User, UserEmail, UserOption
+
+logger = logging.getLogger('sentry.accounts')
+
+
+InvalidEmailResponse = Response({'detail': 'Invalid email'}, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
soo i don't know if this will actually break anything or not, but it seems weird to me to have a global `Response` instance. i _think_ doing something like:
```python
class InvalidEmailResponse(Response):
    def __init__(self):
        super(InvalidEmailResponse, self).__init__(
            {'detail': 'Invalid email'},
            status=status.HTTP_400_BAD_REQUEST
        )
```
and then `return InvalidEmailResponse()` would be less weird, but maybe also still kinda weird. I will defer to Matt or someone else to decide what the best way of dealing with this is.

Reply:
Yeah, don't reuse this instance. It should definitely be created for each Response. It's possible to tack state onto a Response, and this could cause a weird bug to track down.